plugins/optimization-detective/helper.php

```diff
@@ -60,6 +60,108 @@ function od_generate_media_query( ?int $minimum_viewport_width, ?int $maximum_vi
 	}
 }
 
+/**
+ * Gets the reasons why Optimization Detective is disabled for the current response.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return array<string, string> Array of disabled reason codes and their messages.
+ */
+function od_get_disabled_reasons(): array {
+	$disabled_flags = array(
+		'is_search'              => false,
+		'is_embed'               => false,
+		'is_preview'             => false,
+		'is_customize_preview'   => false,
+		'not_get_request'        => false,
+		'no_cache_purge_post_id' => false,
+	);
+
+	// Since there is no predictability in whether posts in the loop will have featured images assigned or not. If a
+	// theme template for search results doesn't even show featured images, then this wouldn't be an issue.
+	if ( is_search() ) {
+		$disabled_flags['is_search'] = true;
+	}
+
+	// Avoid optimizing embed responses because the Post Embed iframes include a sandbox attribute with the value of
+	// "allow-scripts" but without "allow-same-origin". This can result in an error in the console:
+	// > Access to script at '.../detect.js?ver=0.4.1' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
+	// So it's better to just avoid attempting to optimize Post Embed responses (which don't need optimization anyway).
+	if ( is_embed() ) {
+		$disabled_flags['is_embed'] = true;
+	}
+
+	// Skip posts that aren't published yet.
+	if ( is_preview() ) {
+		$disabled_flags['is_preview'] = true;
+	}
+
+	// Since injection of inline-editing controls interfere with breadcrumbs, while also just not necessary in this context.
```

Comment:
```suggestion
	// Disable in Customizer preview since injection of inline-editing controls can interfere with XPath. Optimization is also not necessary in this context.
```